scipy/special/_hyp2f1.pxd

```diff
@@ -196,6 +199,19 @@ cdef inline double complex hyp2f1_complex(
             return result
         # Maximum number of terms 1500 comes from Fortran original.
         return hyp2f1_series(a, b, c, z, 1500, True, EPS)
+    if 0.9 <= modulus_z < 1.1 and zabs(1 - z) >= 0.9 and z.real >= 0:
+        # This condition for applying Euler Transformation (DLMF 15.8.1)
+        # was determined empirically to work better for this case than that
+        # used in the original Fortran implementation for |z| < 0.9,
```

Comment:
General note-one day we will hopefully no longer remember what the heck the "original Fortran" implementation even was because it's been replaced by something so much better... be generally wary of time-sensitive comments like this and try to replace them with things that make sense to someone 5 years from now.

Reply:
> General note-one day we will hopefully no longer remember what the heck the "original Fortran" implementation even was because it's been replaced by something so much better... be generally wary of time-sensitive comments like this and try to replace them with things that make sense to someone 5 years from now.

Great point! I think we have three PRs to go before the Fortran implementation can be expunged and thrown into the memory hole. I'll try to make sure every time-sensitive comment like this is updated by then.